src/scripts/main.js

```diff
@@ -1,3 +1,84 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('tbody');
+const container = document.querySelector('.container');
+const appendRowBtn = document.querySelector('.append-row');
+const removeRowBtn = document.querySelector('.remove-row');
+const appendColumnBtn = document.querySelector('.append-column');
+const removeColumnBtn = document.querySelector('.remove-column');
+
+function updateButtonState() {
+  const rowCount = table.children.length;
+  const columnCount = getUpdatedRows()[0].children.length;
+
+  appendRowBtn.disabled = rowCount >= 10;
+  removeRowBtn.disabled = rowCount <= 2;
+  appendColumnBtn.disabled = columnCount >= 10;
+  removeColumnBtn.disabled = columnCount <= 2;
+}
+
+function getUpdatedRows() {
+  return table.querySelectorAll('tr');
+}
+
+function appendRow() {
+  const newRow = createRow();
+
+  table.appendChild(newRow);
+  updateButtonState();
+}
+
+function removeRow() {
+  table.removeChild(table.lastElementChild);
+  updateButtonState();
+}
+
+function appendColumn() {
+  getUpdatedRows().forEach((row) => {
+    const data = document.createElement('td');
+
+    row.appendChild(data);
+  });
+  updateButtonState();
+}
+
+function removeColumn() {
+  getUpdatedRows().forEach((row) => {
+    row.removeChild(row.lastElementChild);
+  });
+  updateButtonState();
+}
+
+function createRow() {
+  const newRow = document.createElement('tr');
+
+  for (let i = 0; i < getUpdatedRows()[0].children.length; i++) {
+    const data = document.createElement('td');
+
+    newRow.appendChild(data);
+  }
+
+  return newRow;
+}
+
+container.addEventListener('click', (e) => {
+  const targetButton = e.target;
+
+  switch (targetButton.classList.value) {
+    case 'append-row button':
+      appendRow();
+      break;
+    case 'remove-row button':
+      removeRow();
+      break;
+    case 'append-column button':
+      appendColumn();
+      break;
+    case 'remove-column button':
```

Comment:
problem is here. cases with remove button not passed because has different classes order. try to change logic which will be not depend on classes order.
maybe each button can has some unique id
or something like that change.
```suggestion
  switch (true) {
    case target.button.classlist.includes('append-row'):
      appendRow();
      break;
    case target.button.classlist.includes('remove-row'):
      removeRow();
      break;
    casetarget.button.classlist.includes('append-column'):
      appendColumn();
      break;
    casetarget.button.classlist.includes('remove-column'):
```

![Screenshot from 2024-02-17 15-16-27](https://github.com/mate-academy/js_growth_table_DOM/assets/33029864/ef74b395-4e30-45c1-a1ef-76e1bc4dc102)